### PR TITLE
Fixes for snapping issues

### PR DESCRIPTION
**Pull request by alexbruy**
Better comparison of old and new snapped positions.

### CODE REVIEW

File: app/snaputils.cpp [hunk 1]
```diff
@@ -200,7 +199,9 @@ void SnapUtils::setupSnapping()
     }
     case 2:
     {
-      mSnappingUtils.setConfig( mQgsProject->snappingConfig() );
+      QgsSnappingConfig config = mQgsProject->snappingConfig();
+      config.setTolerance( config.tolerance() * InputUtils::calculateDpRatio() );
```
Comment (tomasMizera) [inline app/snaputils.cpp:203]: We need to first check if the tolerance is set in pixels :)

File: app/inpututils.cpp [hunk 2]
```diff
@@ -681,6 +681,35 @@ QgsPointXY InputUtils::transformPoint( const QgsCoordinateReferenceSystem &srcCr
...
+  {
+    return QgsPoint();
+  }
+
+  try
+  {
+    QgsCoordinateTransform ct( srcCrs, destCrs, context );
+    if ( ct.isValid() )
+    {
+      const QgsPoint pt( ct.transform( srcPoint.x(), srcPoint.y() ) );
```
Comment (tomasMizera) [inline app/inpututils.cpp:701]: Don't we drop `Z` and `M` coordinates this way @alexbruy ?

Comment (alexbruy) [reply app/inpututils.cpp]: You are right. Fixed.

**Comment by tomasMizera** [comment]
Hi @alexbruy, can you please link all tickets this should close?

**Comment by alexbruy** [comment]
Should fix #2130 and #2129. Probably also #2132.

**Pull request merged**